scripts/brc4/patch_build/osid_add_organisms.py

```diff
@@ -14,38 +14,54 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+"""Add or update one organism in an OSID server.
+
+You can update an organism by setting a new pattern without setting the gene and transcript starts.
+"""
+
 
 import argparse
-import requests
 import json
+from typing import Dict, Optional
 
+import requests
 
-def get_organism(url, user, key, org_data):
+
+def get_organism(url: str, user: str, key: str, org_data: Dict[str, str]) -> Optional[int]:
+    """Retrieve the organism ID from OSID for a given organism.
+
+    Returns None if no organism could be found with that name.
+    """
     species = org_data["organismName"]
 
     body = {"organismName": species}
 
     get_org = url + "/organisms"
-    result = requests.get(get_org, auth=(user, key), params=body)
+    result = requests.get(get_org, auth=(user, key), params=body, timeout=10)
 
     if result and result.status_code == 200:
         organisms = json.loads(result.content)
-        count = len(organisms)
 
         if len(organisms) == 1:
             return organisms[0]["organismId"]
-        elif len(organisms) == 0:
-            return False
-        else:
-            raise Exception("Found several organisms for the name '%s'" % species)
-    else:
-        print("Error: " + str(result))
+        if len(organisms) == 0:
+            return None
+        raise ValueError(f"Found several organisms for the name '{species}'")
+    print("Error: " + str(result))
+    return None
+
 
+def add_organism(url: str, user: str, key: str, org_data: Dict[str, str]) -> None:
+    """Add a new organism to OSID."""
+
+    # Species is new, check we have all the keys
+    expected = {"organismName", "template", "geneIntStart", "transcriptIntStart"}
+    if set(expected) != set(org_data):
```

Comment:
```suggestion
    if set(expected) != org_data.keys():
```
Equivalent code (tested) and, in my opinion, more readable since it is not clear what `set()` is using from the dictionary.

Reply:
I agree, but I would still prefer to keep the set() on the right side, because it's not clear what comparing a set and a list should output